src/main/java/org/opentripplanner/graph_builder/module/GtfsFeedId.java

```diff
@@ -95,13 +95,14 @@ public GtfsFeedId build() {
      * @param id The feed id
      * @return The cleaned id.
      */
-    protected String cleanId(String id) {
+    private String cleanId(String id) {
       if (id == null || id.trim().length() == 0) {
-        return id;
+        return null;
       }
-      // 1. Underscore is used as an separator by OBA.
-      // 2. Colon is used as an separator in OTP.
-      return id.replaceAll("_", "").replaceAll(":", "");
+      // Colon is used as a separator in OTP so that's why we strip it out - it will confuse the
+      // parsers in the API (but those could be updated if someone is keen to use colons as the feed
+      // id).
+      return id.replaceAll(":", "");
```

Comment:
Use String.replace() instead of String.replaceAll() since the pattern is not a regexp